allennlp/modules/token_vectorizer.py

```diff
@@ -0,0 +1,30 @@
+import torch
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+
+class TokenVectorizer(torch.nn.Module):
+    """
+    A ``TokenVectorizer`` is a ``Module`` that takes as input a tensor with integer ids that have
+    been output from a :class:`~allennutlp.data.TokenIndexer` and outputs a vector per token in the
+    input.  The input typically has shape ``(batch_size, num_tokens)`` or ``(batch_size,
+    num_tokens, num_characters)``, and the output is of shape ``(batch_size, num_tokens,
+    output_dim)``.  The simplest ``TokenVectorizer`` is just an embedding layer, but for
+    character-level input, it could also be some kind of character encoder.
+
+    We add a single method to the basic ``Module`` API: :func:`get_output_dim()`.  This lets us
+    more easily compute output dimensions for the :class:`~allennlp.modules.TokenEmbedder`, which
+    we might need when defining model parameters.
```

Comment:
"such as stacked LSTMs, where we need to know the output dimensions before the layers are called".